stl/inc/memory

```diff
@@ -1504,6 +1556,238 @@ private:
     }
 };
 
+#ifdef __cpp_concepts
+template <class _Ty>
+constexpr _Ty _Div_ceil(_Ty _Val1, _Ty _Val2) noexcept {
+    return (_Val1 % _Val2 == 0) ? (_Val1 / _Val2) : (_Val1 / _Val2 + 1);
+}
+
+template <class _NoThrowIt>
+struct _Uninitialized_rev_destroying_backout { // struct to undo partially constructed ranges in _Uninitialized_xxx
+                                               // algorithms
+    _NoThrowIt _First;
+    _NoThrowIt _Last;
+
+    explicit _Uninitialized_rev_destroying_backout(_NoThrowIt _Dest) : _First(_Dest), _Last(_Dest) {}
+
+    _Uninitialized_rev_destroying_backout(_NoThrowIt _First_, _NoThrowIt _Last_) : _First(_First_), _Last(_Last_) {}
+
+    _Uninitialized_rev_destroying_backout(const _Uninitialized_rev_destroying_backout&) = delete;
+    _Uninitialized_rev_destroying_backout& operator=(const _Uninitialized_rev_destroying_backout&) = delete;
+
+    ~_Uninitialized_rev_destroying_backout() {
+        if (_Last != _First) {
+            do {
+                --_Last;
+                _STD destroy_at(_Last);
+            } while (_Last != _First);
+        }
+    }
+
+    template <class... _Types>
+    void _Emplace_back(_Types&&... _Vals) { // construct a new element at *_Last and increment
+        _Construct_in_place(*_Last, _STD forward<_Types>(_Vals)...);
+        ++_Last;
+    }
+
+    _NoThrowIt _Release() { // suppress any exception handling backout and return _Last
+        _First = _Last;
+        return _Last;
+    }
+};
+
+template <class _Ty>
+void _Reverse_destroy_multidimensional_n(_Ty* const _Arr, const size_t _Size) {
+    if (_Size != 0) {
+        size_t _Idx = _Size;
+        do {
+            --_Idx;
+            // TRANSITION, LLVM-42404
+            if constexpr ((rank_v<_Ty>) > 0) {
+                _Reverse_destroy_multidimensional_n(_Arr[_Idx], extent_v<_Ty>);
+            } else {
+                _Destroy_in_place(_Arr[_Idx]);
+            }
+        } while (_Idx > 0);
+    }
+}
+
+template <class _Ty, size_t _Size>
+void _Uninitialized_copy_multidimensional(const _Ty (&_In)[_Size], _Ty (&_Out)[_Size]) {
+    // TRANSITION, LLVM-42404
+    if constexpr ((rank_v<_Ty>) > 0) {
+        for (size_t _Idx = 0; _Idx < _Size; ++_Idx) {
+            _TRY_BEGIN
+            _Uninitialized_copy_multidimensional(_In[_Idx], _Out[_Idx]);
+            _CATCH_ALL
+            _Reverse_destroy_multidimensional_n(_Out, _Idx);
+            _RERAISE;
+            _CATCH_END
+        }
+    } else {

```

Comment:
I believe you can use an `else if constexpr (_Ptr_copy_cat<_Ty*, _Ty*>::_Really_trivial)` directly here

Reply:
Done